test/tests/functional/pbs_qsub_direct_write.py

```diff
@@ -64,6 +64,126 @@ def test_direct_write_when_job_succeeds(self):
         file_count = len([name for name in os.listdir(
             mapping_dir) if os.path.isfile(os.path.join(mapping_dir, name))])
         self.assertEqual(2, file_count)
+        self.server.expect(JOB, {'Keep_Files': 'doe'}, id=jid)
+
+    def test_direct_write_output_file(self):
+        """
+        submit a sleep job and make sure that the output file
+        are getting directly written directly to the mapped
+        directory when direct_files option is used with o option.
+        """
+        j = Job(TEST_USER)
+        j.set_attributes({ATTR_k: 'do'})
+        j.set_sleep_time(10)
+        sub_dir = DshUtils().mkdtemp(uid=TEST_USER.uid)
+        mapping_dir = DshUtils().mkdtemp(uid=TEST_USER.uid)
+        self.mom.add_config(
+            {'$usecp': self.server.hostname + ':' + sub_dir
+             + ' ' + mapping_dir})
+        self.mom.restart()
+        jid = self.server.submit(j, submit_dir=sub_dir)
+        self.server.log_match(jid + ";Job Run at", max_attempts=10)
+        file_count = len([name for name in os.listdir(
+            mapping_dir) if os.path.isfile(os.path.join(mapping_dir, name))])
```

Comment:
Python has a `filter` function that filters a list based on a condition. 
```
is_file = lambda name : os.path.isfile(os.path.join(mapping_dir, name))
file_count = len(filter(is_file, os.listdir(mapping_dir))
```
I think it is a bit more readable.

Reply:
autopep8 converted this into 
def is_file(name): return os.path.isfile(
            os.path.join(mapping_dir, name))
and complained about having 2 functions , May be i'll keep it the way it is.